src/Functions/generateUUIDv7.cpp

```diff
@@ -158,21 +176,35 @@ class FunctionGenerateUUIDv7Base : public IFunction
     {
         auto col_res = ColumnVector<UUID>::create();
         typename ColumnVector<UUID>::Container & vec_to = col_res->getData();
+        vec_to.resize(input_rows_count);
 
         if (input_rows_count)
         {
-            vec_to.resize(input_rows_count);
-
             /// Not all random bytes produced here are required for the UUIDv7 but it's the simplest way to get the required number of them by using RandImpl
             RandImpl::execute(reinterpret_cast<char *>(vec_to.data()), vec_to.size() * sizeof(UUID));
 
-            /// Note: For performance reasons, clock_gettime is called once per chunk instead of once per UUID. This reduces precision but
-            /// it still complies with the UUID standard.
-            uint64_t timestamp = getTimestampMillisecond();
-            for (UUID & uuid : vec_to)
+            UUIDv7Range uuid_range;
```

Comment:
To make the code less error-prone, let's generally try to keep the implementation and code structure as close as possible to the one in generateSnowflakeId, i.e.
- reserve a range
- then iterate over the rows in the output vector and write the uuids of the reserved range

Reply:
Do you mean to remove "capacity"? Or just the name?

With UUIDv7 we can't have a contiguous range since updating the timestamp means generating a new random counter.

Generating one range per timestamp also gives a performance improvement by removing the overflow check from the inner loop.